zebra-consensus/src/block/subsidy/general.rs

```diff
@@ -102,6 +102,26 @@ pub fn find_output_with_amount(
         .collect()
 }
 
+/// Validate shielded consensus rules as described in [ZIP-213][ZIP-213]
+///
+/// [ZIP-213]: https://zips.z.cash/zip-0213#specification
+pub fn shielded_coinbase(height: Height, network: Network, transaction: &Transaction) -> bool {
```

Comment:
Maybe not return `bool` here but `Result` or something.

Reply:
In https://github.com/ZcashFoundation/zebra/pull/1174 we've been returning `Result`s

Reply:
Yea, `Result` is the way to go. In this context i added a new error `SubsidyError::ShieldedCoinbaseFailed` for this at https://github.com/ZcashFoundation/zebra/pull/1190/commits/e705d2f02e38ca3ade03265bfd19bc19466c2c84